src/WebJobs.Extensions.DurableTask/DurableTaskExtension.cs

```diff
@@ -358,44 +363,71 @@ void IExtensionConfigProvider.Initialize(ExtensionConfigContext context)
             context.AddBindingRule<EntityTriggerAttribute>()
                 .BindToTrigger(new EntityTriggerAttributeBindingProvider(this, connectionName));
 
-            this.taskHubWorker = new TaskHubWorker(this.defaultDurabilityProvider, this, this, loggerFactory: this.loggerFactory, versioningSettings: new VersioningSettings
+            this.getTaskHubWorker = () =>
             {

```

Comment:
I wonder if we should keep some state indicating whether we've run this already. I worry about what could happen if this method gets run twice and we create multiple `TaskHubWorker` objects, creating hard-to-diagnose problems.